packages/openapi3/test/metadata.test.ts

```diff
@@ -0,0 +1,565 @@
+        type: "object",
+        properties: { c: { type: "string" } },
+      },
+      MCreateOrUpdate: {
+        type: "object",
+        properties: {
+          c: { type: "string" },
+          u: { type: "string" },
+        },
+      },
+      MUpdate: {
+        type: "object",
+        properties: {
+          u: { type: "string" },
+        },
+      },
+      MDelete: {
+        type: "object",
+        properties: {
+          d: { type: "string" },
+        },
+      },
+      DRead: {
+        type: "object",
+        properties: {},
+        allOf: [{ $ref: "#/components/schemas/MRead" }],
+      },
+      DQuery: {
+        type: "object",
+        properties: {},
+        allOf: [{ $ref: "#/components/schemas/MQuery" }],
+      },
+      DCreate: {
+        type: "object",
+        properties: {},
+        allOf: [{ $ref: "#/components/schemas/MCreate" }],
+      },
+      DCreateOrUpdate: {
+        type: "object",
+        properties: {},
+        allOf: [{ $ref: "#/components/schemas/MCreateOrUpdate" }],
+      },
+      DUpdate: {
+        type: "object",
+        properties: {},
+        allOf: [{ $ref: "#/components/schemas/MUpdate" }],
+      },
+      DDelete: {
+        type: "object",
+        properties: {},
+        allOf: [{ $ref: "#/components/schemas/MDelete" }],
+      },
+      RRead: {
+        type: "object",
+        properties: { m: { $ref: "#/components/schemas/MRead" } },
+      },
+      RCreate: {
+        type: "object",
+        properties: { m: { $ref: "#/components/schemas/MCreate" } },
+      },
+      RCreateOrUpdate: {
+        type: "object",
+        properties: { m: { $ref: "#/components/schemas/MCreateOrUpdate" } },
+      },
+      RUpdate: {
+        type: "object",
+        properties: { m: { $ref: "#/components/schemas/MUpdate" } },
+      },
+      URead: {
+        type: "object",
+        properties: {
+          e: {
+            anyOf: [{ $ref: "#/components/schemas/MRead" }, { type: "string" }],
+            "x-cadl-name": "M | string",
+          },
+        },
+      },
+      UCreate: {
+        type: "object",
+        properties: {
+          e: {
+            anyOf: [{ $ref: "#/components/schemas/MCreate" }, { type: "string" }],
+            "x-cadl-name": "M | string",
+          },
+        },
+      },
+      UCreateOrUpdate: {
+        type: "object",
+        properties: {
+          e: {
+            anyOf: [{ $ref: "#/components/schemas/MCreateOrUpdate" }, { type: "string" }],
+            "x-cadl-name": "M | string",
+          },
+        },
+      },
+      UUpdate: {
+        type: "object",
+        properties: {
+          e: {
+            anyOf: [{ $ref: "#/components/schemas/MUpdate" }, { type: "string" }],
+            "x-cadl-name": "M | string",
+          },
+        },
+      },
+      UDelete: {
+        type: "object",
+        properties: {
+          e: {
+            anyOf: [{ $ref: "#/components/schemas/MDelete" }, { type: "string" }],
```

Comment:
I was going to say that `DELETE` can't have a body, but then I would have been wrong:

https://www.rfc-editor.org/rfc/rfc7231#section-4.3.5

> A payload within a DELETE request message has no defined semantics; sending a payload body on a DELETE request might cause some existing implementations to reject the request.